packages/react/src/components/CodeSnippet/CodeSnippet.mdx

```diff
@@ -0,0 +1,87 @@
+import { Story, Props, Source, Preview } from '@storybook/addon-docs/blocks';
+import { withKnobs, boolean, select, text } from '@storybook/addon-knobs';
+import * as stories from './CodeSnippet-story.js';
+import CodeSnippet from '../CodeSnippet';
+
+<!-- How do I get <Props> to work?
+ -->
+
+# CodeSnippet
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Skeleton state](#skeleton-state)
+- [Inline](#inline)
+- [Multi-line](#multi-line)
+- [Single-line](#single-line)
+- [Light variant](#light-variant)
+- [Component API](#component-api)
+- [Feedback](#feedback)
+
+## Overview
+
+Code snippets are strings or small blocks of reusable code that can be copied
+and inserted in a code file.
+
+<Preview>
+  <Story id="codesnippet--code-snippet" />
+</Preview>
+
+## Skeleton state
+
+You can use the `CodeSnippetSkeleton` component to render a skeleton variant of
+an code snippet. This is useful to display while content in your code snippet is
+being fetched from an external resource like an API.
+
+<Preview>
+  <Story id="codesnippet--skeleton" />
+</Preview>
+
+## Inline
+
+Inline code snippet's are used for distinguishing a symbol, variable, function
+name or similar small piece of code from it's surrounding text.
+
+<Preview>
+  <Story id="codesnippet--inline" />
+</Preview>
+
+## Multi-line
+
+Multiple line code snippets are used for displaying code with more than one
+line. They're shown in a block and useful for showing classes, functions, blocks
+of styles and the like.
+
+<Preview>
+  <Story id="codesnippet--multiline" />
+</Preview>
+
+## Single-line
+
+Single line code snippets are used when showing code that fits on one line
+without wrapping. Useful for calling out terminal commands, function
+invocations, expressions etc.
+
+<Preview>
+  <Story id="codesnippet--singleline" />
+</Preview>
+
+## Light variant
+
+Use the light prop modifier when using a code snippet on a background other than
+the UI background for that theme. The light prop changes the background color
+token of the code snippet from `field-01` to `field-02`.
+
+<Preview>
+  <Story id="codesnippet--light" />
+</Preview>
+
+## Component API
+
+<Props sort="asc" />
+
+## Feedback
```

Comment:
I think it would be helpful to have a blurb about how we recommend using a library like `clipboard.js` to enable copy functionality since it does not have that ability OOTB